docs/resources/custom_allocation_rules.md

```diff
@@ -90,6 +103,10 @@ resource "datadog_custom_allocation_rules" "order" {
 
 - `rule_ids` (List of String) The list of Custom Allocation Rule IDs, in order. Rules are executed in the order specified in this list. Comes from the `id` field on a `datadog_custom_allocation_rule` resource.
 
+### Optional
+
+- `override_ui_defined_resources` (Boolean) Whether to override UI-defined rules. When set to true, any rules created via the UI that are not defined in Terraform will be deleted and Terraform will be used as the source of truth for rules and their ordering. When set to false, any rules created via the UI that are at the end of order will be kept but will be warned, otherwise an error will be thrown in terraform plan phase. Default is false
```

Comment:
```suggestion
- `override_ui_defined_resources` (Boolean) Whether to override UI-defined rules. When set to true, any rules created through the UI that are not defined in Terraform are deleted and Terraform are used as the source of truth for rules and their ordering. When set to false, any rules created through the UI that are at the end of order are kept but are warned, otherwise an error is thrown in the terraform plan phase. Default value is `false`.
```